nbformat/validator.py

```diff
@@ -299,25 +454,65 @@ def validate(
         relax_add_props=relax_add_props,
         strip_invalid_metadata=strip_invalid_metadata,
     ):
+
         raise error
 
-    if notebook_supports_cell_ids:
-        # if we support cell ids check for uniqueness when validating the whole notebook
-        seen_ids = set()
-        for cell in nbdict["cells"]:
-            cell_id = cell["id"]
-            if cell_id in seen_ids:
-                if repair_duplicate_cell_ids:
-                    # Best effort to repair if we find a duplicate id
-                    cell["id"] = generate_corpus_id()
-                    get_logger().warning(
-                        "Non-unique cell id '{}' detected. Corrected to '{}'.".format(
-                            cell_id, cell["id"]
-                        )
-                    )
-                else:
-                    raise ValidationError(f"Non-unique cell id '{cell_id}' detected.")
-            seen_ids.add(cell_id)
+
+def _try_fix_error(nbdict: Any, version: int, version_minor: int, relax_add_props: bool) -> int:
```

Comment:
the name `try_fix_error` is not very expressive. Maybe `fix_notebook` ? `apply_notebook_fixes` ?
I believe the `try` part refers to the fact that it does its best effort with no guarantee but I'm not sure we would expect an universal fix anyway. Also, when we say `fix`, saying it fixes errors is kinda redundant, and we still don't know what kind of errors it fixes.